test/NITTA/LuaFrontend/Tests.hs

```diff
@@ -22,60 +23,273 @@ module NITTA.LuaFrontend.Tests (
-                    sum(a + t + r)
+                    sum(b + t + r, a)
                 end
-                sum(0)
+                sum(1, 2)
             |]
-        dfg =
-            [ add "a#0" "t#0" ["tmp_1#0"]
-            , add "tmp_1#0" "r#0" ["tmp_0#0"]
-            , loop 0 "tmp_0#0" ["a#0"]
-            , constant (-1) ["r#0"]
-            , constant 2 ["t#0"] :: F String Int
+        (actualName, Lua.FunCall (Lua.NormalFunCall _ (Lua.Args actualArgValue)), Lua.FunAssign _ (Lua.FunBody actualArg _ _)) = findStartupFunction (getLuaBlockFromSources src)
+        expectedValues = ("sum", [Lua.Name "a", Lua.Name "b"], [Lua.Number Lua.IntNum "1", Lua.Number Lua.IntNum "2"])
+     in (actualName, actualArg, actualArgValue) @?= expectedValues
+
+case_find_startup_function_no_args =
+    let src =
+            [__i|
+                function sum()
+                    t = receive()
+                    t = t + 1
+                    send(t)
+                    sum()
+                end
+                sum()
+            |]
+        (actualName, Lua.FunCall (Lua.NormalFunCall _ (Lua.Args actualArgValue)), Lua.FunAssign _ (Lua.FunBody actualArg _ _)) = findStartupFunction (getLuaBlockFromSources src)
+        expectedValues = ("sum", [], [])
+     in (actualName, actualArg, actualArgValue) @?= expectedValues
+
+-- |local a = 2
+case_process_local_assignment_statement =
+    let assignment = Lua.LocalAssign [Lua.Name "a"] (Just [Lua.Number Lua.IntNum "2"])
+        expected = HM.fromList [("a", LuaValueInstance{luaValueInstanceName = "a", luaValueInstanceAssignCount = 0, luaValueInstanceIsConstant = False})]
+        (_, LuaAlgBuilder{algLatestLuaValueInstance}) = runState (processStatement "_" assignment) defaultAlgBuilder
+     in algLatestLuaValueInstance @?= expected
+
+-- |a = 2
+case_process_assignment_statement =
+    let assignment = Lua.Assign [Lua.VarName (Lua.Name "a")] [Lua.Number Lua.IntNum "2"]
+        expected = HM.fromList [("a", LuaValueInstance{luaValueInstanceName = "a", luaValueInstanceAssignCount = 0, luaValueInstanceIsConstant = False})]
+        (_, LuaAlgBuilder{algLatestLuaValueInstance}) = runState (processStatement "_" assignment) defaultAlgBuilder
+     in algLatestLuaValueInstance @?= expected
+
+--case_process_multiple_assignments_statement =
+--    let assignment = Assign [VarName (Name $  "a"), VarName (Name $  "b")] [Number IntNum $  "2", Number FloatNum $  "2.5"]
+--        expected =
+--            Map.fromList
+--                [ ( "a", LuaValueInstance{luaValueInstanceName =  "a", luaValueInstanceAssignCount = 0, luaValueInstanceIsConstant = False})
+--                , ( "b", LuaValueInstance{luaValueInstanceName =  "b", luaValueInstanceAssignCount = 0, luaValueInstanceIsConstant = False})
+--                ]
+--        (_str :: String, AlgBuilder{algLatestLuaValueInstance}) = runState (processStatement ( "_") assignment) defaultAlgBuilder
+--     in algLatestLuaValueInstance @?= expected
```

Comment:
Dead code?

Reply:
yes, this test case was checked here https://github.com/ryukzak/nitta/blob/LuaFrontend_new/test/NITTA/LuaFrontend/Tests.hs#L269.